main/test/flix/Test.Exp.Match.Record.flix

```diff
@@ -0,0 +1,193 @@
+        case { a = { a = { a = { } | _ } | _ } | _ } => true
+    }
+
+    @test
+    def testMatchRecord18(): Bool = match { a = { a = { a = { } }, b = { } }, b = { } } {
+        case { a = { a = { a = { } | _ } | _ } | _ } => true
+    }
+
+    @test
+    def testMatchRecord19(): Bool = match { a = { a = { a = { }, b = { } }, b = { } }, b = { } } {
+        case { a = { a = { a = { } | _ } | _ } | _ } => true
+    }
+
+    @test
+    def testMatchRecord20(): Bool = match { a = { a = { a = { }, b = { } }, b = { } }, b = { } } {
+        case { a = { a = { b = { } | _ } | _ } | _ } => true
+    }
+
+    @test
+    def testMatchRecord21(): Bool = match { a = { a = { a = { }, b = { } }, b = { } }, b = { } } {
+        case { a = { a = { a = { }, b = { } | _ } | _ } | _ } => true
+    }
+
+    @test
+    def testMatchRecord22(): Bool = match { a = { a = { a = { }, b = { } }, b = { } }, b = { } } {
+        case { a = { a = { a = { }, b = { } | _ }, b = { } | _ } | _ } => true
+    }
+
+    @test
+    def testMatchRecord23(): Bool = match { a = { a = { a = { }, b = { } }, b = { } }, b = { } } {
+        case { a = { a = { a = { }, b = { } | _ }, b = { } | _ }, b = { } | _ } => true
+    }
+
+    @test
+    def testMatchRecord24(): Bool = match { a = { a = { a = { }, b = { } }, b = { } }, b = { x = A.A } } {
+        case { a = { a = { a = { } | _ } | _ }, b = { x = A.A } } => true
+        case { a = { a = { a = { } | _ } | _ }, b = { x = A.B } } => false
+    }
+
+    @test
+    def testMatchRecord25(): Bool = match { a = { a = { a = { }, b = { b = A.B } }, b = { } }, b = { x = A.A } } {
+        case { a = { a = { a = { } | _ } | _ }, b = { x = A.A } } => true
+        case { a = { a = { a = { } | _ } | _ }, b = { x = A.B } } => false
+    }
+
+    @test
+    def testMatchRecord26(): Bool = match { a = { a = { a = { a = A.B }, b = { b = A.B } }, b = { } }, b = { x = A.A } } {
+        case { a = { a = { a = { a = A.A } | _ } | _ }, b = { x = A.A } } => false
+        case { a = { a = { a = { a = A.A } | _ } | _ }, b = { x = A.B } } => false
+        case { a = { a = { a = { a = A.B } | _ } | _ }, b = { x = A.A } } => true
+        case { a = { a = { a = { a = A.B } | _ } | _ }, b = { x = A.B } } => false
+    }
+
+    @test
+    def testMatchRecord27(): Bool = match { a = A.A, b = A.A } {
+        case { b = A.A, a = A.A } => true
+        case { b = A.A, a = A.B } => false
+        case { b = A.B, a = A.A } => false
+        case { b = A.B, a = A.B } => false
+    }
+
+    @test
+    def testMatchRecord28(): Bool = match { a = { a = { }, b = A.B }, b = A.A } {
+        case { a = { a = { } | _ } | _ } => true
+    }
+    @test
+    def testMatchRecord29(): Bool = match { a = true, b = true } {
+        case { a | r } => a and r.b
+    }
+
+    @test
+    def testMatchRecord30(): Bool = match { a = false, a = true } {
+        case { a = x, a = y } => (not x) and y
+    }
+
```

Comment:
It is good to see so many test cases, but I feel like some more systematic is needed (i.e. by naming each test case set better).

For example along the lines:

- matching on distinct fields
- matching with "rest row"
- matching in random order
- matching with duplicate labels
- matching with nested records